crates/turborepo-lib/src/engine/mod.rs

```diff
@@ -277,11 +278,34 @@ impl Engine<Built> {
             })
         }
 
+        validation_errors.extend(self.validate_interactive(experimental_ui));
+
         match validation_errors.is_empty() {
             true => Ok(()),
             false => Err(validation_errors),
         }
     }
+
+    // Validates that UI is setup if any interactive tasks will be executed
+    fn validate_interactive(&self, experimental_ui: bool) -> Vec<ValidateError> {
```

Comment:
This will be removed once the UI is the default. The check will come down to just "is `turbo` itself running in an interactive terminal"

Reply:
That's a cool protection actually. Turbo stops you from accidentally running interactive processes in CI